pkg/operator/periodicbackupcontroller/periodicbackupcontroller.go

```diff
@@ -318,3 +334,114 @@ func newCronJob() (*batchv1.CronJob, error) {
 
 	return obj.(*batchv1.CronJob), nil
 }
+
+func deployBackupServerDaemonSet(cr backupv1alpha1.Backup, endpoints []string) *appv1.DaemonSet {
+	ds := appv1.DaemonSet{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      backupServerDaemonSet,
+			Namespace: operatorclient.TargetNamespace,
+			Labels: map[string]string{
+				backupDSLabelKey: backupDSLabelValue,
+			},
+		},
+		Spec: appv1.DaemonSetSpec{
+			Selector: &v1.LabelSelector{
+				MatchLabels: map[string]string{
+					backupDSLabelKey: backupDSLabelValue,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: v1.ObjectMeta{
+					Labels: map[string]string{
+						backupDSLabelKey: backupDSLabelValue,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Tolerations: []corev1.Toleration{
+						{
+							Key:    masterNodeSelector,
+							Effect: corev1.TaintEffectNoSchedule,
+						},
+					},
+					NodeSelector: map[string]string{masterNodeSelector: ""},
+					Volumes: []corev1.Volume{
+						{Name: etcdDataDirVolName, VolumeSource: corev1.VolumeSource{
+							HostPath: &corev1.HostPathVolumeSource{
+								Path: etcdDataDirVolPath,
+								Type: ptr.To(corev1.HostPathUnset)}}},
+
+						{Name: etcdConfigDirVolName, VolumeSource: corev1.VolumeSource{
+							HostPath: &corev1.HostPathVolumeSource{
+								Path: etcdConfigDirVolPath}}},
+
+						{Name: etcdAutoBackupDirVolName, VolumeSource: corev1.VolumeSource{
+							HostPath: &corev1.HostPathVolumeSource{
+								Path: etcdAutoBackupDirVolPath}}},
+
+						{Name: etcdCertDirVolName, VolumeSource: corev1.VolumeSource{
+							HostPath: &corev1.HostPathVolumeSource{
+								Path: "/etc/kubernetes/static-pod-resources/etcd-certs"}}},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:    etcdBackupServerContainerName,
+							Image:   os.Getenv("OPERATOR_IMAGE"),
+							Command: []string{"cluster-etcd-operator", "backup-server"},
+							Env: []corev1.EnvVar{
```

Comment:
don't we need all other env variables from the static pod?

Reply:
no, not at all, we only need 3 envVars needed for the `TLSConfig`